twisted/spread/pb.py

```diff
@@ -981,14 +1005,16 @@ def proto_error(self, requestID, fail):
     def sendDecRef(self, objectID):
         """(internal) Send a DECREF directive.
         """
-        self.sendCall("decref", objectID)
+        self.sendCall(b"decref", objectID)
 
     def proto_decref(self, objectID):
         """(internal) Decrement the reference count of an object.
 
         If the reference count is zero, it will free the reference to this
         object.
         """
+        if isinstance(objectID, unicode):
+            objectID = objectID.encode('utf8')
         refs = self.localObjects[objectID].decref()
```

Comment:
since localObjects is a public API, maybe it should be replaced with a dict like implementation which converts the objectID on the fly . I guess that external code might also want to access it and the forget to do the conversion..

or maybe it should be private and it needs to be deprecated as a public API